<?xml version="1.0"?>
<launch>
  <!-- launch file parameters -->
  <arg name="namespace"     default="left" />
  <arg name="grid_spacing"  default="15.94" />
  <arg name="samples"       default="50" />
  <arg name="pausetime"     default="1.0" />
  <arg name="use_weights"   default="True" />
  <arg name="debug"         default="True" />
  
  <!-- Camera driver configuration at start -->
  <node name="driver_dynconfig" pkg="dynamic_reconfigure" type="dynparam" args="set_from_parameters ensenso_driver" ns="camera">
    <!-- Capture parameters -->
    <param name="FrontLight"        type="bool"   value="false" />
    <param name="Projector"         type="bool"   value="false" />
    <param name="TargetBrightness"  type="int"    value="70" />
    <!-- Streaming parameters -->    
    <param name="Cloud"   type="bool"   value="false" />
    <param name="Images"  type="bool"   value="true" />
  </node>
  
  <!-- Image view -->
  <node name="left_image_view" pkg="image_view" type="image_view">
    <remap from="image" to="/camera/left/image_raw" />
  </node>
  
  <!-- Joint trajectory controller -->
  <include file="$(find denso_control)/launch/joint_trajectory_controller.launch">
    <arg name="namespace" value="$(arg namespace)"/>
  </include>
  
  <!-- camera-robot calibration script -->
  <node name="camera_robot_calibration" pkg="axxb_data_collection" type="camera_robot_calibration.py" output="screen" required="True">
    <param name="robot_name"    type="string" value="$(arg namespace)" />
    <param name="debug"         type="bool"   value="$(arg debug)"/>
    <param name="grid_spacing"  type="double" value="$(arg grid_spacing)" />
    <param name="grid_spacing"  type="double" value="$(arg grid_spacing)" />
    <param name="use_weights"   type="bool"   value="$(arg use_weights)"/>
    <param name="pausetime"     type="double" value="$(arg pausetime)"/>
    <param name="samples"       type="int"    value="$(arg samples)"/>
    <param name="maxdist"       type="double" value="1.2"/> <!-- meters -->
    <param name="maxtilt"       type="double" value="15"/>  <!-- degrees -->
  </node>
</launch>
